.github/workflows/test.yaml

```diff
@@ -48,30 +53,54 @@ jobs:
                 exit 1
                 ;;
         esac
+        case "${{ matrix.version }}" in
+            pristine)
+                rm -f test-build
+                ;;
+            test)
+                touch test-build
+                ;;
+        esac
 
     - name: Build snapd snap
       uses: snapcore/action-build@v1
       with:
         snapcraft-channel: 8.x/stable
 
+    - name: Build test snapd snap
+      uses: snapcore/action-build@v1
+      with:
+        snapcraft-channel: 8.x/stable
+
     - name: Check built artifact
       run: |
-        unsquashfs snapd*.snap meta/snap.yaml usr/lib/snapd/
-        if cat squashfs-root/meta/snap.yaml | grep -q "version:.*dirty.*"; then
-          echo "PR produces dirty snapd snap version"
-          cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
-          exit 1
-        elif cat squashfs-root/usr/lib/snapd/info | grep -q "VERSION=.*dirty.*"; then
-          echo "PR produces dirty internal snapd info version"
-          cat squashfs-root/usr/lib/snapd/info
-          cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
-          exit 1
-        fi
+        for snap in snapd*.snap; do
+          case "${snap}" in
+            snapd_1337.*.snap)
+              # Skip test build
+              ;;
+            *)
+              echo "Checking ${snap}"
+              rm -rf squashfs-root
+              unsquashfs "${snap}" meta/snap.yaml usr/lib/snapd/
+              if grep -q "version:.*dirty.*" squashfs-root/meta/snap.yaml; then
+                echo "PR produces dirty snapd snap version"
+                cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
+                exit 1
+              elif grep -q "VERSION=.*dirty.*" squashfs-root/usr/lib/snapd/info; then
+                echo "PR produces dirty internal snapd info version"
+                cat squashfs-root/usr/lib/snapd/info
+                cat squashfs-root/usr/lib/snapd/dirty-git-tree-info.txt
+                exit 1
+              fi
+              ;;
+          esac
+        done
 
     - name: Uploading snapd snap artifact
       uses: actions/upload-artifact@v3
       with:
-        name: snap-files-${{ matrix.toolchain }}
+        name: snap-files-${{ matrix.toolchain }}-${{ matrix.version }}
```

Comment:
is it possible to add a comment somewhere to explain what are those snap artifacts used for?

Reply:
something like Maciej did below